esmvaltool/preprocessor/_multimodel.py

```diff
@@ -155,6 +155,8 @@ def _put_in_cube(template_cube,
 
 def _to_datetime(delta_t, unit_type):
     """Convert to a datetime point"""
+    leap_correction = delta_t / (365. * 4.)
+    delta_t = delta_t + leap_correction
```

Comment:
Why is the leap correction applied depending on the units and not on the calendar?

Reply:
ha! I knew you gonna ask this -- good question -- it's because all dates have been projected to 365-day calendar in advance of calling _to_datetime(); this is done when looking for overlaps or when assembling the full-time-axis

Reply:
Which function is doing this projection?

Reply:
_time_pts_by_calendar(cube) and _get_overlap(cubes)

Reply:
But those functions only consider `360_day` as an exception, but there are many other calendars (`no_leap`, `all_leap`, ...) which could potentially be problematic (especially when processing very long time periods, where the error could accumulate).

Would it make sense on the long term to think about using a function like [this one](http://unidata.github.io/netcdf4-python/#netCDF4.num2date)?
It automatically converts NetCDF time coordinates to datetime objects taking into account units and calendar.

Reply:
Oh yeah absolutely! If only I knew about this a month ago :) I will try use this, but am not gonna change stuff just now...see if you find issues with the current namelist, let's get this PR in and then I will modify the dates handling using this on a future branch. What do you think?

Reply:
That's fine. :+1: 
I'll open an issue just that we don't forget about it.

Reply:
sweet, thanks @mattiarighi